components/org.wso2.carbon.uuf.common.foundation/src/main/fragments/uuf-client/public/uuf-client/uuf-client.js

```diff
@@ -0,0 +1,264 @@
+        if (!zone) {
+            throw new UUFClientException("Zone '" + zoneName + "' does not exists in current DOM.");
+        }
+
+        var url = contextPath + "/fragments/" + fragmentFullyQualifiedName;
+        $.ajax({
+                   url: url,
+                   type: "GET",
+                   data: templateFillingObject,
+                   success: function (data, textStatus, jqXHR) {
+                       pushContent(data, zone, mode);
+                   },
+                   error: function (jqXHR, textStatus, errorThrown) {
+                       throw new UUFClientException("Error occurred while retrieving fragment '"
+                                                    + fragmentFullyQualifiedName + "' from '" + url
+                                                    + "'.", errorThrown, jqXHR.status);
+                   }
+               });
+    };
+
+    /**
+     * Renders the specified embedded Handlebars template and pushes to the given zone according to the given mode.
+     *
+     * @param {string} templateId id of the handlebars template
+     * @param {?Object} templateFillingObject data for the template
+     * @param {string} zoneName name of the zone to push
+     * @param {string} mode mode to use
+     */
+    UUFClient.renderTemplate = function (templateId, templateFillingObject, zoneName, mode) {
+        if (!templateId) {
+            throw new UUFClientException("Template name cannot be null or empty.");
+        }
+        checkNullOrEmpty(zoneName, mode);
+
+        var source = $("#" + templateId).html();
+        if (source) {
+            UUFClient.renderTemplateString(source, templateFillingObject, zoneName, mode);
+        } else {
+            throw new UUFClientException("Template '" + templateId + "' not found");
+        }
+    };
+
+    /**
+     * Retrieves the Handlebars template from the specified URL and renders and pushes it to the given zone according
+     * to the given mode.
+     *
+     * @param {string} templateUrl url of the template
+     * @param {?Object} templateFillingObject data for the template
+     * @param {string} zoneName name of the zone to push
+     * @param {string} mode mode to use
+     */
+    UUFClient.renderTemplateUrl = function (templateUrl, templateFillingObject, zoneName, mode) {
+        if (!templateUrl) {
+            throw new UUFClientException("Template url cannot be null or empty.");
+        }
+        checkNullOrEmpty(zoneName, mode);
+
+        $.ajax({
+                   url: templateUrl,
+                   type: "GET",
+                   success: function (data, textStatus, jqXHR) {
+                       var contentType = jqXHR.getResponseHeader("content-type") || "";
+                       if (contentType.indexOf('text/x-handlebars-template') <= -1) {
+                           throw new UUFClientException(
+                               "Incorrect content type received. Expected: text/x-handlebars-template,"
```

Comment:
Change error message to `"Response content type is incorrect. Found '" + jqXHR.getResponseHeader("content-type") + "' instead of 'text/x-handlebars-template'."`.